tests/marine/test_marine.py

```diff
@@ -289,6 +297,136 @@ def test_http_packet_filter_and_parse(marine_or_marine_pool: Union[Marine, Marin
         packet_encapsulation=encap_consts.ENCAP_ETHERNET,
         bpf_filter=bpf_filter,
         display_filter=display_filter,
+        macros=None,
+        expected_passed=True,
+        expected_output=expected_output,
+    )
+
+
+def test_tcp_packet_filter_and_parse_with_macro(
```

Comment:
I think another test is required. 
The case is when a macro changes (for example you filter packet 1 with `macro1 = [field1, field2]`, and then you filter packet 2 with `macro1 = [field1, field2, field3]` or something, just to verify that the cache works properly :)

Another case could be `macro1 = [field1, field3]` (removing a field), etc.